app/controllers/api/metrics_controller.rb

```diff
@@ -124,7 +124,7 @@ def find_service
   end
 
   def find_metric
-    @metric = @service.metrics.find(params[:id])
+    @metric = @service.all_metrics.find(params[:id])
```

Comment:
We could probably just do Metric.find and verify in Ruby that it belongs to the Service either directly or via BackendApiConfig